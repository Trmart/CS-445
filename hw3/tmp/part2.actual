.   .   .   Child: 0  Call: dog [line: 51]
.   .   .   Child: 0  Const 1184 of type int [line: 39]
.   .   .   Child: 0  Const 13 of type int [line: 50]
.   .   .   Child: 0  Const 220 of type int [line: 40]
.   .   .   Child: 0  Const 3 of type int [line: 34]
.   .   .   Child: 0  Const 666 of type int [line: 10]
.   .   .   Child: 0  Const 'a' [line: 45]
.   .   .   Child: 0  Const 'a' [line: 46]
.   .   .   Child: 0  Const 'a' [line: 47]
.   .   .   .   Child: 0  Const 'c' [line: 47]
.   .   .   .   Child: 0  Const true of type bool [line: 47]
.   .   .   .   Child: 0  Const true of type bool [line: 51]
.   .   .   .   Child: 0  Id: c [line: 35]
.   .   .   Child: 0  Id: x [line: 22]
.   .   .   Child: 0  Id: y [line: 23]
.   .   .   Child: 0  Id: z [line: 4]
.   .   .   Child: 0  Id: z [line: 49]
.   .   .   Child: 0  Id: z [line: 5]
.   .   .   Child: 0  Op: + [line: 35]
.   Child: 0  Parm: x of type int [line: 18]
.   Child: 0  Parm: z of array of type bool [line: 28]
.   .   Child: 0  Var: c of type char [line: 34]
.   .   Child: 0  Var: x of type int [line: 20]
.   .   Child: 0  Var: z of array of type int [line: 3]
.   .   Child: 1  Assign: <= [line: 22]
.   .   Child: 1  Break [line: 38]
.   Child: 1  Compound [line: 14]
.   Child: 1  Compound [line: 19]
.   Child: 1  Compound [line: 2]
.   Child: 1  Compound [line: 26]
.   Child: 1  Compound [line: 29]
.   Child: 1  Compound [line: 33]
.   Child: 1  Compound [line: 9]
.   .   .   Child: 1  Const 1210 of type int [line: 39]
.   .   .   Child: 1  Const 2620 of type int [line: 22]
.   .   .   Child: 1  Const 2620 of type int [line: 4]
.   .   .   Child: 1  Const 284 of type int [line: 40]
.   .   .   Child: 1  Const 2924 of type int [line: 23]
.   .   .   .   Child: 1  Const 3 of type int [line: 47]
.   .   .   .   Child: 1  Const 6 of type int [line: 35]
.   .   Child: 1  If [line: 4]
.   .   Child: 1  Return [line: 10]
.   .   Child: 1  Return [line: 15]
ERROR(23): Symbol 'y' is not declared.
ERROR(36): Symbol 'z' is already declared at line 3.
Func: ant returns type int [line: 1]
Number of errors: 2
Number of warnings: 4
.   .   Sibling: 10  Call: elk [line: 51]
.   .   Sibling: 1  Assign: <= [line: 23]
.   .   .   .   Sibling: 1  Const 3 of type int [line: 51]
.   .   .   Sibling: 1  Const 'b' [line: 46]
.   .   .   Sibling: 1  Const 'b' [line: 47]
Sibling: 1  Func: bat returns type void [line: 8]
.   Sibling: 1  Parm: y of type int [line: 18]
.   .   Sibling: 1  Return [line: 5]
.   .   Sibling: 1  Var: d of type char [line: 35]
.   .   Sibling: 1  While [line: 39]
Sibling: 2  Func: cat returns type int [line: 13]
.   .   Sibling: 2  If [line: 40]
.   .   .   Sibling: 2  Op: + [line: 47]
.   .   Sibling: 2  Var: z of array of type int [line: 36]
.   .   Sibling: 3  Call: zebu [line: 42]
Sibling: 3  Func: dog returns type void [line: 18]
.   .   .   Sibling: 3  Op: chsign [line: 47]
.   .   Sibling: 4  Call: dog [line: 44]
Sibling: 4  Func: dog returns type char [line: 26]
.   .   Sibling: 5  Call: dog [line: 45]
Sibling: 5  Func: elk returns type void [line: 28]
.   .   Sibling: 6  Call: dog [line: 46]
Sibling: 6  Func: main returns type void [line: 32]
.   .   Sibling: 7  Call: dog [line: 47]
.   .   Sibling: 8  Call: dog [line: 49]
.   .   Sibling: 9  Call: elk [line: 50]
WARNING(20): The variable 'x' seems not to be used.
WARNING(34): The variable 'c' seems not to be used.
WARNING(35): The variable 'd' seems not to be used.
WARNING(3): The variable 'z' seems not to be used.
