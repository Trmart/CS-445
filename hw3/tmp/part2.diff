1,13c1,60
< ERROR(20): Symbol 'x' is already declared at line 18.
< ERROR(26): Symbol 'dog' is already declared at line 18.
< ERROR(35): '+' requires operands of type int but lhs is of type char.
< ERROR(42): Symbol 'zebu' is not declared.
< ERROR(47): '+' requires operands of type int but lhs is of type char.
< ERROR(47): Unary 'chsign' requires an operand of type int but was given type bool.
< ERROR(5): Cannot return an array.
< Number of errors: 7
< Number of warnings: 4
< WARNING(28): The variable 'z' seems not to be used.
< WARNING(35): The variable 'd' seems not to be used.
< WARNING(49): Variable 'z' may be uninitialized when used here.
< WARNING(4): Variable 'z' may be uninitialized when used here.
---
> .   .   .   Child: 0  Const 1184 [line: 39]
> .   .   .   Child: 0  Const 220 [line: 40]
> .   .   .   Child: 0  Const 3 [line: 34]
> .   .   .   Child: 0  Const 666 [line: 10]
> .   .   .   Child: 0  Const 'a' [line: 45]
> .   .   .   Child: 0  Const 'a' [line: 46]
> .   .   .   Child: 0  Const 'a' [line: 47]
> .   .   .   .   Child: 0  Const 'c' [line: 47]
> .   .   .   .   Child: 0  Id: c [line: 35]
> .   .   .   Child: 0  Id: x [line: 22]
> .   .   .   Child: 0  Id: y [line: 23]
> .   .   .   Child: 0  Id: z [line: 4]
> .   .   .   Child: 0  Id: z [line: 5]
> .   .   .   Child: 0  Op: + [line: 35]
> .   Child: 0  Parm: x of type int [line: 18]
> .   Child: 0  Parm: z of array of type bool [line: 28]
> .   .   Child: 0  Var: c of type char [line: 34]
> .   .   Child: 0  Var: x of type int [line: 20]
> .   .   Child: 0  Var: z of array of type int [line: 3]
> .   .   Child: 1  Assign: <= [line: 22]
> .   .   Child: 1  Break [line: 38]
> .   Child: 1  Compound [line: 14]
> .   Child: 1  Compound [line: 19]
> .   Child: 1  Compound [line: 2]
> .   Child: 1  Compound [line: 26]
> .   Child: 1  Compound [line: 29]
> .   Child: 1  Compound [line: 33]
> .   Child: 1  Compound [line: 9]
> .   .   .   Child: 1  Const 1210 [line: 39]
> .   .   .   Child: 1  Const 2620 [line: 22]
> .   .   .   Child: 1  Const 2620 [line: 4]
> .   .   .   Child: 1  Const 284 [line: 40]
> .   .   .   Child: 1  Const 2924 [line: 23]
> .   .   .   .   Child: 1  Const 3 [line: 47]
> .   .   .   .   Child: 1  Const 6 [line: 35]
> .   .   Child: 1  If [line: 4]
> .   .   Child: 1  Return [line: 10]
> .   .   Child: 1  Return [line: 15]
> Func: ant returns type int [line: 1]
> .   .   Sibling: 1  Assign: <= [line: 23]
> .   .   .   Sibling: 1  Const 'b' [line: 46]
> .   .   .   Sibling: 1  Const 'b' [line: 47]
> Sibling: 1  Func: ( returns type void [line: 8]
> .   Sibling: 1  Parm: y of type int [line: 18]
> .   .   Sibling: 1  Return [line: 5]
> .   .   Sibling: 1  Var: d of type char [line: 35]
> .   .   Sibling: 1  While [line: 39]
> Sibling: 2  Func: cat returns type int [line: 13]
> .   .   Sibling: 2  If [line: 40]
> .   .   .   Sibling: 2  Op: + [line: 47]
> .   .   Sibling: 2  Var: z of array of type int [line: 36]
> .   .   Sibling: 3  Call: zebu [line: 42]
> Sibling: 3  Func: ( returns type void [line: 18]
> .   .   Sibling: 4  Call: dog [line: 44]
> Sibling: 4  Func: dog returns type char [line: 26]
> .   .   Sibling: 5  Call: dog [line: 45]
> Sibling: 5  Func: ( returns type void [line: 28]
> .   .   Sibling: 6  Call: dog [line: 46]
> Sibling: 6  Func: ( returns type void [line: 32]
> .   .   Sibling: 7  Call: dog [line: 47]
