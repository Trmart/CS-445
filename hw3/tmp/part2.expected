ERROR(20): Symbol 'x' is already declared at line 18.
ERROR(26): Symbol 'dog' is already declared at line 18.
ERROR(35): '+' requires operands of type int but lhs is of type char.
ERROR(42): Symbol 'zebu' is not declared.
ERROR(47): '+' requires operands of type int but lhs is of type char.
ERROR(47): Unary 'chsign' requires an operand of type int but was given type bool.
ERROR(5): Cannot return an array.
Number of errors: 7
Number of warnings: 4
WARNING(28): The variable 'z' seems not to be used.
WARNING(35): The variable 'd' seems not to be used.
WARNING(49): Variable 'z' may be uninitialized when used here.
WARNING(4): Variable 'z' may be uninitialized when used here.
